### PR TITLE
feat: added PR template for submitters

### PR DESCRIPTION
**Pull request by borntorock**
Partial fixes https://github.com/jenkins-x/jx-docs/issues/3116

This PR includes the github Pull Request template as reference for the PR submitters.

### CODE REVIEW

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -0,0 +1,24 @@
+# Description
+
+Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.
+
+Fixes # (issue)
+
+## Type of change
```
Comment (ankitm123) [inline .github/PULL_REQUEST_TEMPLATE.md:7]: This seems excessive - the type field in the commit message is enough.

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+
+Please delete options that are not relevant.
+
+- [ ] Bug fix (non-breaking change which fixes an issue)
+- [ ] New feature (non-breaking change which adds functionality)
+- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
+- [ ] This change requires a documentation update
```
Comment (ankitm123) [inline .github/PULL_REQUEST_TEMPLATE.md:14]: Dont think this is relevant

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+
+# Checklist:
+
+- [ ] My code follows the [style guidelines](https://jenkins-x.io/community/code/) of this project
+- [ ] I have performed a self-review of my own code
+- [ ] I have made corresponding changes to the documentation in this PR itself, as per [docs guidelines](https://jenkins-x.io/community/documentation/)
+- [ ] I have mentioned the appropriate tag (either of feat, docs, fix) in the commit message and PR title for semantic checks to pass.
+- [ ] I have added unit tests that prove my fix is effective or that my feature works
```
Comment (ankitm123) [inline .github/PULL_REQUEST_TEMPLATE.md:22]: Not required

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+- [ ] Bug fix (non-breaking change which fixes an issue)
+- [ ] New feature (non-breaking change which adds functionality)
+- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
+- [ ] This change requires a documentation update
+
+# Checklist:
+
+- [ ] My code follows the [style guidelines](https://jenkins-x.io/community/code/) of this project
+- [ ] I have performed a self-review of my own code
+- [ ] I have made corresponding changes to the documentation in this PR itself, as per [docs guidelines](https://jenkins-x.io/community/documentation/)
```
Comment (ankitm123) [inline .github/PULL_REQUEST_TEMPLATE.md:20]: Not sure what this is

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+
+- [ ] Bug fix (non-breaking change which fixes an issue)
+- [ ] New feature (non-breaking change which adds functionality)
+- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
+- [ ] This change requires a documentation update
+
+# Checklist:
+
+- [ ] My code follows the [style guidelines](https://jenkins-x.io/community/code/) of this project
+- [ ] I have performed a self-review of my own code
```
Comment (ankitm123) [inline .github/PULL_REQUEST_TEMPLATE.md:19]: Probably not relevant to docs

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+Please delete options that are not relevant.
+
+- [ ] Bug fix (non-breaking change which fixes an issue)
+- [ ] New feature (non-breaking change which adds functionality)
+- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
+- [ ] This change requires a documentation update
+
+# Checklist:
+
+- [ ] My code follows the [style guidelines](https://jenkins-x.io/community/code/) of this project
```
Comment (ankitm123) [inline .github/PULL_REQUEST_TEMPLATE.md:18]: Code?

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 3 above)
Comment (borntorock) [reply .github/PULL_REQUEST_TEMPLATE.md:22]: okay

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 4 above)
Comment (borntorock) [reply .github/PULL_REQUEST_TEMPLATE.md:20]: okay, removed this point too!

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 7]
```diff
@@ -0,0 +1,11 @@
+# Description
+
+Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.
+
+Fixes # (issue)
+
+# Checklist:
+
+- [ ] I have mentioned the appropriate tag (either of feat, docs, fix) in the commit message and PR title for semantic checks to pass.
```
Comment (ankitm123) [inline .github/PULL_REQUEST_TEMPLATE.md:9]: Would be good to add some information abt how to write the commit message here: https://jenkins-x.io/community/documentation/workflow/#commit-and-push-your-changes and link this doc to the PR template. For now, we can link this: https://jenkins-x.io/community/code/#the-commit-message

Also change tag to type: https://github.com/angular/angular.js/blob/master/DEVELOPERS.md#commit-message-format -> `type(scope): subject`

Comment (borntorock) [reply .github/PULL_REQUEST_TEMPLATE.md]: Done

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 8]
```diff
@@ -0,0 +1,11 @@
+# Description
+
+Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.
+
+Fixes # (issue)
+
+# Checklist:
+
+- [ ] I have mentioned the appropriate type(scope), as referenced [here](https://jenkins-x.io/community/code/#the-commit-message) in the commit message and PR title for the semantic checks to pass.
+- [ ] Any dependent changes have already been merged
```
Comment (ankitm123) [inline .github/PULL_REQUEST_TEMPLATE.md:10]: What is this about? Should we keep this?

Comment (borntorock) [reply .github/PULL_REQUEST_TEMPLATE.md:10]: not sure, what do you think? I added this considering the current experience Kara's PR was on hold to merge before your PR of versioning. In these cases, I think good to have in description.

Comment (borntorock) [reply .github/PULL_REQUEST_TEMPLATE.md:10]: OTOH, shall we also mention to sign the commit?

Comment (ankitm123) [reply .github/PULL_REQUEST_TEMPLATE.md:10]: > not sure, what do you think? I added this considering the current experience Kara's PR was on hold to merge before your PR of versioning. In these cases, I think good to have in description.

Ah that makes sense :+1: 

> OTOH, shall we also mention to sign the commit?

Yes! Link to the doc which talks abt it.

**Pull request merged**